test/test_compile.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import os.path
+import pytest
+
+from otk.command import compile
+
+
+@pytest.mark.parametrize(
+    "parser, arguments, dry_run, input_data, output_data, sys_exit_code, log_message",
+    [
+        # "GENERATE" is a placeholder to append "tmp_path" from pytest
+        # dry_run must not generate an output
+        (
+            None,
```

Comment:
This seems to be "None" for all inputs? Or did I just not look careful enough?

Reply:
Well the parameter itself is not even used in the function - yes, I could drop it from the test for now...

Reply:
@supakeen I can as well remove the parameter "parser" from `compile` (and `validate` and `_process()` how it is called here)
Probably this will never be used?

Reply:
@mvo5 btw. also `arguments.input` is always `"GENERATE"` because I wanted to test the stdin case - but I don't know how to mock `/dev/stdin`. I could mock `sys.stdin` but not `/dev/stdin` - any ideas? 🤔
(without rewriting the whole functionality of otk 😅)

Reply:
Ah yes, I added the `parser` argument I think because I had ideas about testing but they weren't very good. Let's remove that argument.

Perhaps there is a better way than my POSIX extension hacks to do the stdin and stdout handling. I did it this way so the code could just use `pathlib` everywhere. It'd be possible I guess to go with file-descriptors instead.

Reply:
If you don't mind I'll also change this within this PR as this would be needed to mock stdin I guess

Reply:
Using sys.{stdin,stdout} seems fine with me, is there any downside? It's also easy to mock via pytest

Reply:
Let's keep as is for now and do the stdin/stdout testing in a followup. This way we can also rework it a bit, I think using filedescriptors instead of path has some benefits (including that it will make the sys.std{in,out} tests simpler and that tit will have less TOCTOU issues (which are not very important here)

Reply:
I don't think this would be TOCTOU...
please check my suggestion (next force push)